explorer/interpreter/action_stack.cpp

```diff
@@ -155,16 +165,37 @@ auto ActionStack::FinishAction(Nonnull<const Value*> result)
     case Action::Kind::ExpressionAction:
     case Action::Kind::LValAction:
     case Action::Kind::PatternAction:
-      PopScopes();
+      destroy_actions_ = PopScopes();
       SetResult(result);
+      break;
+    case Action::Kind::CleanUpAction:
```

Comment:
Is CleanUpAction valid here? It doesn't call SetResult, so it seems like it shouldn't be used this way.

Reply:
You are right! Removed!

Reply:
Sorry, I am wrong.... I need the case, without it I can't compile because the compiler checks if all action types are present in the switch statement

Reply:
Continuing as normal is still incorrect behavior. How about moving it up to the group with RecursiveAction ("This kind of Action cannot produce results: ")?

Reply:
done